### PR TITLE
Feature/task4747

### PR DESCRIPTION
**Pull request by msamandi**
[XXXX-<Title> - Please use the Work Item number and Title as PR Name, not subtasks]

#### 📲 What

Refactor multiple REST Controller to a single By-Resource REST Controller

#### 🤔 Why

Why it's needed, background context.

#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class MenuControllerV2 {
+
+  @Autowired private MenuMapper menuMapper;
+  @Autowired private MenuQueryService menuQueryService;
+
+  @GetMapping(value = "/{id}")
+  @Operation(
+      tags = "Menu",
+      summary = "Get a menu",
+      security = @SecurityRequirement(name = "bearerAuth"),
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java:36]: You can safely remove this securityRequirement annotation

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping(path = "/v2/menu", produces = "application/json; charset=utf-8")
+@RestController
+public class MenuControllerV2 {
+
+  @Autowired private MenuMapper menuMapper;
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java:29]: Possibly convert to use Constructor injection, if that's the pattern we have decided to follow?

File: java/src/test/java/com/amido/stacks/workloads/menu/domain/MenuHelper.java [hunk 3]
```diff
@@ -15,7 +15,7 @@ public static List<Menu> createMenus(int count) {
     return menuList;
   }
 
-  public static Menu createMenu(int counter) {
+  public static McreateMenu(int counter) {
```
Comment (sdavis-amido) [inline java/src/test/java/com/amido/stacks/workloads/menu/domain/MenuHelper.java]: Looks odd - does this even compile?

**Review by sdavis-amido — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java (hunk 1 above)
Comment (msamandi) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java:36]: done

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping(path = "/v1/menu", produces = MediaType.APPLICATION_JSON_VALUE + " ; charset=utf-8")
+@RestController
+public class MenuController {
+
+  private CreateMenuHandler createMenuHandler;
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java:56]: I suspect IntelliJ is telling you all of these can be `final`?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(
+    path = "/v1/menu/{id}/category",
+    produces = MediaType.APPLICATION_JSON_VALUE + "; charset=utf-8")
+public class CategoryController {
+
+  private CreateCategoryHandler createCategoryHandler;
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java:39]: I suspect IntelliJ is telling you all of these can be final?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping(
+    path = "/v1/menu/{id}/category/{categoryId}/items",
+    produces = MediaType.APPLICATION_JSON_VALUE + "; charset=utf-8")
+@RestController
+public class ItemController {
+
+  private CreateItemHandler createItemHandler;
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java:39]: I suspect IntelliJ is telling you all of these can be final?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping(path = "/v2/menu", produces = "application/json; charset=utf-8")
+@RestController
+public class MenuControllerV2 {
+
+  @Autowired
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java:28]: Remove autowired and make it constructor injection, and also make the fields `final`

**Review by sdavis-amido — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java (hunk 2 above)
Comment (msamandi) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java]: this was my question ?!

Comment (sdavis-amido) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java]: Yes, let's go for CI :-)

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java (hunk 4 above)
Comment (msamandi) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java:56]: yes it does in big Yellow ...

Comment (msamandi) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java:56]: fixed

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java (hunk 5 above)
Comment (msamandi) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java:39]: yes done

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java (hunk 6 above)
Comment (msamandi) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java:39]: yeah

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java (hunk 7 above)
Comment (msamandi) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v2/MenuControllerV2.java:28]: done

**Comment by msamandi** [comment]
reviewed

**Pull request merged**